handlers.py

```diff
@@ -59,69 +64,71 @@ async def registration(message: types.Message):
+        await bot.send_message(chat_id=queries[message.from_user.id][-1],
+                               text=f"User {message.from_user.first_name} @{message.from_user.username} with number {database.get_contact(message.from_user.id)[0][0]} is here:\n {json_data['response']['GeoObjectCollection']['featureMember'][1]['GeoObject']['metaDataProperty']['GeocoderMetaData']['text']}")
+        await bot.send_message(chat_id=queries[message.from_user.id][-1],
                                text=' '.join([str(message['location']['latitude']), str(message['location']['longitude'])]))
-        tracking_id.popleft()
-
-    @dp.message_handler(content_types=["contact"])
-    async def give_contact(message: types.Contact):
-        await bot.send_message(chat_id=tracking_id[0],
-                               text=f"Unfortunately, your request has been rejected but you can call {message['contact']['phone_number']}")
-        tracking_id.popleft()
+        queries[message.from_user.id].pop()
+        if len(queries[message.from_user.id]) != 0:
+            keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
+            button1 = types.KeyboardButton("Yes", request_location=True)
+            button2 = types.KeyboardButton("No", request_contact=True)
+            keyboard.add(button1, button2)
+            await bot.send_message(message.from_user.id,
+                                   text=f"User {database.get_name(queries[message.from_user.id][-1])[0][0]} @{database.get_username(queries[message.from_user.id][-1])[0][0]} with number {database.get_contact(queries[message.from_user.id][-1])[0][0]} wants to track you, are you agree?",
+                                   reply_markup=keyboard)
+
+
+    @dp.message_handler(lambda message: message.text == "No")
+    async def give_contact(message: types.Message):
+        await bot.send_message(chat_id=queries[message.from_user.id][-1],
+                               text=f"Unfortunately,user {database.get_name(message.from_user.id)[0][0]} {database.get_username(message.from_user.id)[0][0]} with number {database.get_contact(message.from_user.id)[0][0]} rejected your request, try to make a call!")
+        queries[message.from_user.id].pop()
+        if len(queries[message.from_user.id]) != 0:
+            keyboard = types.ReplyKeyboardMarkup(resize_keyboard=True, one_time_keyboard=True)
+            button1 = types.KeyboardButton("Yes", request_location=True)
+            button2 = types.KeyboardButton("No")
+            keyboard.add(button1, button2)
+            await bot.send_message(message.from_user.id,
+                                   text=f"User {database.get_name(queries[message.from_user.id][-1])[0][0]} @{database.get_username(queries[message.from_user.id][-1])[0][0]} with number {database.get_contact(queries[message.from_user.id][-1])[0][0]} wants to track you, are you agree?",
+                                   reply_markup=keyboard)
```

Comment:
look at this part as well in terms of the comments that have been left above

Reply:
here repeated code is ineitable, cause we can't take a piece of code and name it somehow in order to copy and paste it's name. that's why we have to use such a construction. again, we can do without it but it was made for the sake of users as they can get a lot of requests from a lot of friends/colleagues and all of them must be counted well